client/src/components/ratingsReviews/RatingsAndReviews.jsx

```diff
@@ -217,21 +220,59 @@ class RatingsAndReviews extends React.Component {
     });
   }
 
+  updateRatingsList(filteredList) {
+    this.setState({
+      filteredRatings: filteredList
+    }, () => {
+
+      console.log('TEST1:', this.state.filteredRatings);
+    });
+  }
+
   handleRatingProgressFilter(event) {
-    event.preventDefault();
-    let starFilter = parseInt(event.target.id);
-    console.log('TEST3', this.state[starFilter]);
-    this.state[starFilter] ?
+    let filteredRatings = [];
+    let starFilter = event.target.id;
+
+    if (this.state[starFilter]) {
+      console.log('Star Filter is true');
       this.setState({
-        starFilter: false
+        [starFilter]: false
       }, () => {
-        this.getReviews(this.state.sort, this.state.count, this.props.selectedProduct.id);
-      })
-      : this.setState({
-        starFilter: true
+        console.log('Ratings before cut:', filteredRatings);
+        this.state.reviews.forEach((review) => {
```

Comment:
Remember that `setState` is async. It usually works out automatically in a render method, since React will just re-render when state updates, but don't count on the expected values being there when you're doing pre-render calculations.